src/compiler/checker.ts

```diff
@@ -19434,7 +19434,7 @@ namespace ts {
                         // the type param can be compared with itself in the target (with the influence of its constraint to match other parts)
                         // For example, if `T extends 1 | 2` and `U extends 2 | 3` and we compare `T & U` to `T & U & (1 | 2 | 3)`
                         const constraint = getEffectiveConstraintOfIntersection(source.flags & TypeFlags.Intersection ? (source as IntersectionType).types: [source], !!(target.flags & TypeFlags.Union));
-                        if (constraint && everyType(constraint, c => c !== source)) { // Skip comparison if expansion contains the source itself
+                        if (constraint && everyType(constraint, c => !isTypeIdenticalTo(c, source))) { // Skip comparison if expansion contains the source itself

```

Comment:
If I'm normalizing, I would probably just normalize `constraint`, but given there is work that `isRelatedTo` does before normalizing it itself, maybe that's not right.

Reply:
I decided to just normalize the constraint; after following the code, there's no way that any of those early checks would apply, because they don't apply to intersections.